src/spi2parallel.cpp

```diff
@@ -0,0 +1,81 @@
+#include "spi2parallel.h"
+
+Spi2Parallel::Spi2Parallel(uint8_t pinCS, uint8_t deviceCount, uint8_t bitOrder)
+    : _pinCS(pinCS)
+    , _deviceCount(deviceCount)
+    , _bitOrder(bitOrder)
+    , _deviceChain(new uint8_t[_deviceCount - 1])
```

Comment:
_deviceChain(new uint8_t[_deviceCount - 1]) - if _deviceCount == 1 we will have array size == 0. Remove '-1'.

Reply:
:open_mouth:  Indeed! Good catch!

How could it even pass a test?!

Reply:
Oooppss, thanks.

Function `new  uint8_t [0 ]` create a pointer to memory with 0 size. 
Unfortunately, everything worked in the Arduino IDE.